### PR TITLE
Fixed type mismatch in sync_commands

### PR DESCRIPTION
**Pull request by wurgo**
## Summary

Previously, the 'type' and 'guild_id' variables couldn't match because they weren't of the same type. I've now changed the types so they can compare with the correct types.

## Checklist

- [x] If code changes were made then they have been tested.
    - [ ] I have updated the documentation to reflect the changes.
- [ ] If `type: ignore` comments were used, a comment is also left explaining why
- [x] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, typehinting, examples, ...)

### CODE REVIEW

File: discord/bot.py [hunk 1]
```diff
@@ -577,8 +577,8 @@ async def sync_commands(
             guild_permissions: List = []
 
             for i in commands:
-                cmd = find(lambda cmd: cmd.name == i["name"] and cmd.type == i["type"] and cmd.guild_ids is not None
-                                       and (i["guild_id"]) in cmd.guild_ids, self.pending_application_commands)
+                cmd = find(lambda cmd: cmd.name == i["name"] and str(cmd.type) == str(i["type"]) and cmd.guild_ids is not None
```
Comment (Dorukyum) [inline discord/bot.py:580]: Aren't `cmd.type` and `i["type"]` both integers?
```suggestion
                cmd = find(lambda cmd: cmd.name == i["name"] and cmd.type == i["type"] and cmd.guild_ids is not None
```

Comment (wurgo) [reply discord/bot.py:580]: Yes, but I didn't want to take any chances, sorry in advance :)

Comment (Dorukyum) [reply discord/bot.py:580]: No need to be sorry, it's fine if other devs approve this. I just want to avoid unnecessary conversions to keep the speed.

Comment (Lulalaby) [reply discord/bot.py:580]: I think @wurgo's change is fine.

File: discord/bot.py [hunk 2]
```diff
@@ -576,8 +576,8 @@ async def sync_commands(
...
+                                       and (int(i["guild_id"])) in cmd.guild_ids, self.pending_application_commands)
```
Comment (krittick) [inline discord/bot.py:580]:
```suggestion
                                       and int(i["guild_id"]) in cmd.guild_ids, self.pending_application_commands)
```

Remove unnecessary parenthesis - this was actually a missing int() per here: https://discord.com/channels/881207955029110855/881735314987708456/939338589412655145

File: discord/bot.py (hunk 1 above)
Comment (krittick) [reply discord/bot.py:579]: I'm struggling to see a situation where cmd.type and i["type"] would ever not be integers based on the latest command registration rewrite, and agree with @Dorukyum that we should avoid unnecessary conversions where possible.

**Review by Dorukyum — APPROVED** [review]

**Review by Lulalaby — APPROVED** [review]

**Comment by Dorukyum** [comment]
It seems we kept the str conversion, I though the last commit reverted that when I took a glance :pain:

**Comment by wurgo** [comment]
> It seems we kept the str conversion, I though the last commit reverted that when I took a glance :pain:

sorry lol

**Pull request merged**